server/core/src/https/mod.rs

```diff
@@ -429,568 +128,306 @@ pub async fn create_https_server(
-                        if let Err(e) = server_result {
-                            error!(
-                                "Failed to accept via listener on address {:?} -> {:?}",
-                                &address, e
-                            );
-                        }
-                    }
-                };
-                info!("Stopped HTTPSAcceptorActor");
-            })
+async fn server_loop(
+    tls_param: TlsConfiguration,
+    addr: SocketAddr,
+    app: IntoMakeServiceWithConnectInfo<Router, SocketAddr>,
+) -> Result<(), std::io::Error> {
+    let mut tls_builder = SslAcceptor::mozilla_intermediate_v5(SslMethod::tls())?;
+    let mut app = app;
+    tls_builder
+        .set_certificate_file(tls_param.chain.clone(), SslFiletype::PEM)
+        .map_err(|err| {
+            std::io::Error::new(
+                ErrorKind::Other,
+                format!("Failed to create TLS listener: {:?}", err),
+            )
+        })?;
+    tls_builder
+        .set_private_key_file(tls_param.key.clone(), SslFiletype::PEM)
+        .map_err(|err| {
+            std::io::Error::new(
+                ErrorKind::Other,
+                format!("Failed to create TLS listener: {:?}", err),
+            )
+        })?;
+    tls_builder.check_private_key().map_err(|err| {
+        std::io::Error::new(
+            ErrorKind::Other,
+            format!("Failed to create TLS listener: {:?}", err),
+        )
+    })?;
+    let acceptor = tls_builder.build();
+    let listener = TcpListener::bind(addr).await?;
+
+    let protocol = Arc::new(Http::new());
+    let mut listener =
+        hyper::server::conn::AddrIncoming::from_listener(listener).map_err(|err| {
+            std::io::Error::new(
+                ErrorKind::Other,
+                format!("Failed to create listener: {:?}", err),
+            )
+        })?;
+    loop {
+        if let Some(Ok(stream)) = poll_fn(|cx| Pin::new(&mut listener).poll_accept(cx)).await {
+            let acceptor = acceptor.clone();
+            let svc = tower::MakeService::make_service(&mut app, &stream);
+            tokio::spawn(handle_conn(acceptor, stream, svc, protocol.clone()));
         }
-        None => {
-            // Create without https
-            let mut listener = (&address).to_listener().map_err(|e| {
-                error!("Failed to convert to Listener -> {:?}", e);
-            })?;
+    }
+}
+// #[instrument(name = "handle-connection", level = "debug", skip_all)]
+/// This handles an individual connection.
+async fn handle_conn(
+    acceptor: SslAcceptor,
+    stream: AddrStream,
+    svc: ResponseFuture<Router, SocketAddr>,
+    protocol: Arc<Http>,
+) -> Result<(), hyper::Error> {
+    #[allow(clippy::expect_used)]
+    let mut tls_stream = SslStream::new(
+        #[allow(clippy::expect_used)]
+        // because if this fails well, then we're going to bail!
+        Ssl::new(acceptor.context()).expect("Failed to build SSL session"),
```

Comment:
Then return an error rather then the panic?

Reply:
I think this will panic if a plaintext conn comes in to a tls port ....

Reply:
I've replaced these with `std::io::Error` handlers and that should sort it